Sources/STTextView/STTextView+Delete.swift

```diff
@@ -83,7 +83,7 @@ extension STTextView {
             }
         }
 
-        if textRanges.isEmpty {
+        if textRanges.isEmpty || !shouldChangeText(in: textRanges, replacementString: nil) {
```

Comment:
It should be `""`, because later it's called with `""` as well

```suggestion
        if textRanges.isEmpty || !shouldChangeText(in: textRanges, replacementString: "") {
```